pkg/logql/metrics.go

```diff
@@ -314,23 +334,36 @@ func RecordVolumeQueryMetrics(
 		latencyType = latencyTypeSlow
 	}
 
+	rangeType := "range"
+	if step == 0 {
+		rangeType = "instant"
+	}
+
 	level.Info(logger).Log(
 		"latency", latencyType,
 		"query_type", queryType,
 		"query", query,
 		"query_hash", HashedQuery(query),
+		"start", start.Format(time.RFC3339Nano),
+		"end", end.Format(time.RFC3339Nano),
+		"start_delta", time.Since(start),
```

Comment:
Q: for my understanding:

why `start_delta` and `end_delta`? And both calculated from `time.Since` (meaning the time where it is logging in `metrics.go`?. Failing to understand what's the relation here.

why can't be just `delta: end-start`?

Reply:
We use the same thing in our query logs. These figures useful for knowing the difference in time between the start of the query range and when the query began (or more strictly, when it completed - which is normally not long after).

This is used to see how many queries are being executed for some time-range other than from now to now-6h, for example.